uaclient/entitlements/fips.py

```diff
@@ -49,6 +31,33 @@ class FIPSCommonEntitlement(repo.RepoEntitlement):
     help_doc_url = "https://ubuntu.com/security/certifications#fips"
     _incompatible_services = ["livepatch"]
 
+    @property
+    def conditional_packages(self):
+        """
+        Dictionary of conditional packages to be installed when
+        enabling FIPS services. For example, if we are enabling
+        FIPS services in a machine that has openssh-client installed,
+        we will perform two actions:
+
+        1. Upgrade the package to the FIPS version
+        2. Install the correspinding hmac version of that package.
+        """
+        conditional_packages = ["strongswan", "strongswan-hmac"]
+
+        series = util.get_platform_info().get("series", "")
+        # On Focal, we don't have the openssh hmac packages.
+        # Therefore, we will not try to install them during
+        # when enabling any FIPS service
+        if series != "focal":
```

Comment:
Given that this will likely be focal++, I think we may want to say the following:
```suggestion
        if series in ('xenial', 'bionic')
```

Reply:
Yes, using `apt-cache policy` was my first option here. My only concern with it is that it could cause incomplete FIPS installs. For example, suppose that for some reason, the `openssh-client-hmac` package disappears from the pocket. If we are using the apt-cache policy on a xenial enable. We will enable FIPS but not install that package. I know this is a corner case, but that's why I have not used that approach

Reply:
Yeah this is true. The case I was thinking of was in Xenial if security team does remove openssh-client-hmac it would only be because the necessary complement of "-hmac" functionality would be released in the openssh-client deb delivered in the FIPS xenial-updates pocket. Making packages disappear from a launchpad PPA is hard to do and would have to be intentional by the security team. Given that FIPS is/was certified with a certain set of packages I don't think that can ever change anyway. But that said, I think we shouldn't spend the cycles trying to make it more dynamic, since each LTS version of FIPS should be a frozen package set anyway due to the certification constraints for it. If it becomes more dynamic in the "J" LTS release, then we can revisit. What you have is good for me.